### PR TITLE
Pack IV

### PR DESCRIPTION
**Pull request by olorin**
! @markhibberd @erikd-ambiata
/jury approve @markhibberd @ambiata-ci @olorin

### CODE REVIEW

File: test/Test/Tinfoil/Arbitrary.hs [hunk 1]
```diff
@@ -95,3 +95,9 @@ instance Arbitrary InvocationCount where
 
 instance Arbitrary InvocationField where
   arbitrary = InvocationField <$> arbitrary <*> arbitrary
+
+instance Arbitrary FixedField where
+  arbitrary = FixedField <$> choose (0, 2^(32 :: Int) - 1)
```
Comment (erikd-ambiata) [inline test/Test/Tinfoil/Arbitrary.hs]: That `choose` thing is identical to `arbitrary`.

You're right that the arbitrary instance for `Int` is weird and doesn't give anything *near* a uniform distribution, bit the instance for `Word32` is more than fine. If it was by code I'd just change that to `arbitrary` and maybe add a comment that the type is `Word32` and that the `Arbitrary` instance for that type is known to be sane.

**Comment by erikd-ambiata** [comment]
Up to you whether you want to update that arbitrary instance.

:+1:

File: test/Test/Tinfoil/Arbitrary.hs (hunk 1 above)
Comment (olorin) [inline test/Test/Tinfoil/Arbitrary.hs]: Ah cool, I didn't know that. Will fix, thanks.

File: test/Test/Tinfoil/Arbitrary.hs [hunk 2]
```diff
@@ -95,3 +95,10 @@ instance Arbitrary InvocationCount where
 
 instance Arbitrary InvocationField where
   arbitrary = InvocationField <$> arbitrary <*> arbitrary
+
+-- Word32 arbitrary is uniform over the range (unlike Int)
+instance Arbitrary FixedField where
+  arbitrary = FixedField <$> arbitrary
```
Comment (markhibberd) [inline test/Test/Tinfoil/Arbitrary.hs]: arbitraryBoundedIntegral if you want to be explicit.

**Comment by markhibberd** [comment]
👍  

/jury approve

File: test/Test/Tinfoil/Arbitrary.hs (hunk 2 above)
Comment (erikd-ambiata) [inline test/Test/Tinfoil/Arbitrary.hs]: TIL about `arbitraryBoundedIntegral`. Explicit is always better than implicit.

Comment (olorin) [inline test/Test/Tinfoil/Arbitrary.hs]: SGTM, thanks.

**Pull request merged**